src/app/Services/card.service.ts

```diff
@@ -1,55 +1,56 @@
 
-  public addProductCard(product: Product): void {
-    const cardList = this._cardListSubject$.value;
-    if (!cardList.some(element => element.id === product.id)) {
-      this._cardListSubject$.next([
-        ...cardList, {
-          'id': product.id,
-          'quantity': 0,
-        },
-      ]);
+    public addProductCard(product: Product): void {
+        const cardList = this._cardListSubject$.value;
+        if (!cardList.some(element => element.id === product.id)) {
+            this._cardListSubject$.next([
+                ...cardList, {
+                    'id': product.id,
+                    'quantity': 0,
+                },
+            ]);
+        }
+        this.incrementProductCard(product);
     }
-    this.incrementProductCard(product);
-  }
 
-  public substractProductCard(card: Card): void {
-    const cardList = this._cardListSubject$.value.map(element =>
-      element.id === card.id ? { ...element, quantity: element.quantity - 1 } : element,
-    );
-    this._cardListSubject$.next(cardList);
-    const cardElement = cardList.find(element => element.id === card.id);
-    if (cardElement && cardElement.quantity <= 0) {
-      this.deleteFromCard(cardElement.id);
+    public substractProductCard(card: Card): void {
+        const cardList = this._cardListSubject$.value.map(element =>
+            element.id === card.id ? {...element, quantity: element.quantity - 1} : element,
+        );
+        this._cardListSubject$.next(cardList);
+        const cardElement = cardList.find(element => element.id === card.id);
+        if (cardElement && cardElement.quantity <= 0) {
+            this.deleteFromCard(cardElement.id);
+        }
     }
-  }
 
-  public incrementProductCard(product: any): void {
-    const cardList = this._cardListSubject$.value;
-    this._cardListSubject$.next(
-      cardList.map(element => {
-        if (element.id === product.id) {
-          return { ...element, quantity: element.quantity + 1 };
-        }
-        return element;
-      }),
-    );
-  }
+    public incrementProductCard(product: any): void {
+        const cardList = this._cardListSubject$.value;
+        this._cardListSubject$.next(
+            cardList.map(element => {
+                if (element.id === product.id) {
+                    return {...element, quantity: element.quantity + 1};
+                }
+                return element;
+            }),
+        );
+    }
 
-  public deleteFromCard(id: number): void {
-    const currentCard = this._cardListSubject$.value;
-    this._cardListSubject$.next(currentCard.filter(product => product.id !== id));
-  }
+    public deleteFromCard(id: number): void {
+        const currentCard = this._cardListSubject$.value;
+        this._cardListSubject$.next(currentCard.filter(product => product.id !== id));
+    }
 
-  public get cardList$(): Observable<Card[]> {
-    return this._cardListSubject$.asObservable();
-  }
+    public get cardList$(): Signal<Card[]> {
```

Comment:
```suggestion
    public get cardList(): Signal<Card[]> {
```

Un signal n'est pas suffixé par `$`